docs/source/advanced/multi-modules.mdx

```diff
@@ -118,3 +118,8 @@ The `bidirectional` parameter is experimental and not compatible with <a href="h
 
 </ExperimentalFeature>
 
+## Optional optimizations
+
+In Android projects, to further improve build times you can consider making your schema module a JVM module instead of an Android library. Since Apollo Kotlin is multiplatform, and Android is capable of depending on JVM-only modules, you can benefit from reducing the amount of work needed to compile the schema module. This is true because JVM modules build faster than Android libraries even with identical source code.  
+This is often a good idea because Apollo often happens to be on the critical path of compiling your app. This happens when all the feature modules that use Apollo need to wait for all the type-safe code to be generated before they themselves can start building too.
```

Comment:
Do we have an authoritative link for this? Android docs, AGP source code, something else?

Reply:
My search so far hasn't provided me with some authoritative source talking about this.
It's true that I wrote this simply from my own experiences, but that is definitely not any concrete evidence. I asked around a bit to try and see if others know of any such source, I will try to come back here if I find something about it.